tasks/species_typing/task_tbprofiler_output_parsing.wdl

```diff
@@ -285,117 +552,130 @@ task tbprofiler_output_parsing {
 
         # reported mutation by tb-profiler, all confering resistance by WHO criteria
         for dr_variant in results_json["dr_variants"]: 
+          genes_reported.append(dr_variant["gene"])
           if "annotation" in dr_variant:
-            try:  # sometimes annotation is an empty list
-              if dr_variant["annotation"][0]["who_confidence"] == "":
-                confidence = "No WHO annotation"
-              else:
-                confidence = dr_variant["annotation"][0]["who_confidence"]
-            except:
-              confidence = "No WHO annotation"
+            # empty dictionary in the case where multiple drugs are identified
+            drugs_to_row = {}
+            # case: annotation field is present but it's an empty list, expert rule is applied directly 
+            if len(dr_variant["annotation"]) == 0:
+              row = variant_to_row(dr_variant)
+              row["confidence"] = "No WHO annotation"
+              row["looker_interpretation"] = apply_expert_rules(dr_variant["nucleotide_change"], dr_variant["protein_change"], dr_variant["gene"], dr_variant["type"], "looker")
+              row["mdl_interpretation"] = apply_expert_rules(dr_variant["nucleotide_change"], dr_variant["protein_change"], dr_variant["gene"], dr_variant["type"], "MDL")
+              row["rationale"] = "Expert rule applied"
+
+              # iterate through any gene-associated drugs
+              for identified_drug in dr_variant["gene_associated_drugs"]:
+                if identified_drug not in drugs_to_row:
+                  drugs_to_row[identified_drug] = {"other_variant": dr_variant, "who_confidence": row["confidence"], "drug": identified_drug, "nucleotide_change": dr_variant["nucleotide_change"], "protein_change": dr_variant["protein_change"], "gene": dr_variant["gene"], "type": dr_variant["type"]}
```

Comment:
consider breaking these long dictionaries into multiple lines to make it more readable and easier to edit in the future